components/board/genericlinux/data.go

```diff
@@ -144,53 +144,41 @@ func getPwmChipDefs(pinDefs []PinDefinition) (map[string]pwmChipData, error) {
 
 	// Now, look for all chips whose names we found.
 	pwmChipsInfo := map[string]pwmChipData{}
+	var found bool
+	const sysPrefix = "/sys/class/pwm"
+	files, err := os.ReadDir(sysPrefix)
+	if err != nil {
+		return nil, err
+	}
+
 	for chipName := range pwmChipNames {
-		found := false
-
-		// The boards we support might store their PWM devices in several different locations
-		// within /sys/devices. If it turns out that every board stores its PWM chips in a
-		// different location, we should rethink this. Maybe a breadth-first search over all of
-		// /sys/devices/? but for now, this is good enough.
-		directoriesToSearch := []string{
-			// Example boards which use each location:
-			// Jetson Orin AGX       BeagleBone AI64                     Intel UP 4000
-			"/sys/devices/platform", "/sys/devices/platform/bus@100000", "/sys/devices/pci0000:00",
-		}
-		for _, baseDir := range directoriesToSearch {
-			// For exactly one baseDir, there should be a directory at <baseDir>/<chipName>/pwm/,
-			// which contains a single sub-directory whose name is mirrored in /sys/class/pwm.
-			// That's the one we want to use.
-			chipDir := fmt.Sprintf("%s/%s/pwm", baseDir, chipName)
-			files, err := os.ReadDir(chipDir)
+		found = false
+		for _, file := range files {
+			if !strings.HasPrefix(file.Name(), "pwmchip") {
+				continue
+			}
+
+			// look at symlinks to find the correct chip
+			symlink, err := os.Readlink(filepath.Join(sysPrefix, file.Name()))
 			if err != nil {
-				continue // This was the wrong directory; try the next baseDir.
+				golog.Global().Errorw(
+					"cannot find symlinks for", "chip", file.Name(), "err:", err)
+				continue
 			}
 
-			// We've found what looks like the right place to look for things! Now, find the name
-			// of the chip that should be mirrored in /sys/class/pwm/.
-			for _, file := range files {
-				if !(strings.Contains(file.Name(), "pwmchip") && file.IsDir()) {
-					continue
-				}
+			if strings.Contains(symlink, chipName) {
 				found = true
-				chipPath := fmt.Sprintf("/sys/class/pwm/%s", file.Name())
-
+				chipPath := filepath.Join("/sys/class/pwm", file.Name())
```

Comment:
One more time, I'd replace `"/sys/class/pwm"` with `sysPrefix`. That way, if it ever needs to change (e.g., we encounter some new board that puts things in a different place and need to take that into account), we only need to change the value in one place.